### ISSUE TITLE
Hashrate ~15% of benchmark rate

### ISSUE THREAD

**Issue by muhfugen** [ethereum-mining/ethminer#455]
Can anyone explain why the actual hashrate when mining would be significantly less than when benchmarking? The system is a VM with a single E5-2660 v2 core, 2GB RAM, a GTX980 on Win10 1703 with ethminer 0.12.0. While both mining and benchmarking, I can see that the GPU utilization is at 100%.

**ethminer -U -M**
```
 cu  15:55:35|main      Using grid size 8192 , block size 128
Benchmarking on platform: CUDA
Preparing DAG for block #0
  i  15:55:35|CUDA0     set work; seed: #00000000, target: #000000000000
  i  15:55:35|CUDA0     Initialising miner...
Warming up...
 cu  15:55:36|CUDA0     Using device: GeForce GTX 980  (Compute 5.2)
 cu  15:55:37|CUDA0     Generating DAG for GPU # 0
CUDA#0: 0%
CUDA#0: 6%
CUDA#0: 13%
CUDA#0: 19%
CUDA#0: 25%
CUDA#0: 31%
CUDA#0: 38%
CUDA#0: 44%
CUDA#0: 50%
CUDA#0: 56%
CUDA#0: 63%
CUDA#0: 69%
CUDA#0: 75%
CUDA#0: 81%
CUDA#0: 88%
CUDA#0: 94%
Trial 1... 19932910
Trial 2... 20142730
Trial 3... 19934904
Trial 4... 19932910
Trial 5... 19930916
min/mean/max: 19930916/19974874/20142730 H/s
inner mean: 20003514 H/s
```

**ethminer.exe --farm-recheck 200 -U -S us1.ethermine.org:4444 -FS us2.ethermine.org:4444 -O MY_ADDRESS.NODE**
```
 cu  15:57:23|main      Using grid size 8192 , block size 128
  i  15:57:23|main      Connecting to stratum server us1.ethermine.org:4444
  i  15:57:23|stratum   Connected to stratum server us1.ethermine.org : 4444
  i  15:57:23|stratum   Starting farm
  i  15:57:23|stratum   Subscribed to stratum server
  i  15:57:23|stratum   Authorized worker 0x4927EBDD4163932C76f005E45E3A4C50cFE42FaD.ESXI1
  i  15:57:23|stratum   Received new job #2d304c59
  i  15:57:23|CUDA0     set work; seed: #88615dc9, target: #0000000112e0
  i  15:57:23|CUDA0     Initialising miner...
  m  15:57:23|main      Speed   0.00 Mh/s    gpu/0  0.00  [A0+0:R0+0:F0] Time: 00:00  i  15:57:24|stratum   Received new job #d47047f8 cu  15:57:27|CUDA0     Using device: GeForce GTX 980  (Compute 5.2)


 cu  15:57:27|CUDA0     Generating DAG for GPU # 0
CUDA#0: 0%
CUDA#0: 3%
CUDA#0: 6%
CUDA#0: 8%
CUDA#0: 11%
CUDA#0: 14%
CUDA#0: 17%
CUDA#0: 19%
CUDA#0: 22%
CUDA#0: 25%
CUDA#0: 28%
CUDA#0: 31%
CUDA#0: 33%
CUDA#0: 36%
CUDA#0: 39%
CUDA#0: 42%
CUDA#0: 44%
CUDA#0: 47%
CUDA#0: 50%
CUDA#0: 53%
CUDA#0: 56%
CUDA#0: 58%
CUDA#0: 61%
CUDA#0: 64%
CUDA#0: 67%
CUDA#0: 69%
CUDA#0: 72%
CUDA#0: 75%
CUDA#0: 78%
CUDA#0: 81%
CUDA#0: 83%
CUDA#0: 86%
CUDA#0: 89%
CUDA#0: 92%
CUDA#0: 94%
CUDA#0: 97%
  i  15:57:37|CUDA0     set work; seed: #88615dc9, target: #0000000112e0
  i  15:57:37|stratum   Received new job #c17092c4  m  15:57:37|main      Speed   0.00 Mh/s    gpu/0  0.00  [A0+0:R0+0:F0] Time: 00:00

  i  15:57:39|CUDA0     set work; seed: #88615dc9, target: #0000000112e0
  m  15:57:39|main      Speed   0.00 Mh/s    gpu/0  0.00  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:39|main      Speed   0.00 Mh/s    gpu/0  0.00  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:39|main      Speed   0.00 Mh/s    gpu/0  0.00  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:39|main      Speed   0.00 Mh/s    gpu/0  0.00  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:40|main      Speed   0.00 Mh/s    gpu/0  0.00  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:41|main      Speed   4.20 Mh/s    gpu/0  4.20  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:41|main      Speed   4.20 Mh/s    gpu/0  4.20  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:41|main      Speed   4.20 Mh/s    gpu/0  4.20  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:42|main      Speed   4.20 Mh/s    gpu/0  4.20  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:42|main      Speed   3.15 Mh/s    gpu/0  3.15  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:43|main      Speed   2.80 Mh/s    gpu/0  2.80  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:43|main      Speed   2.80 Mh/s    gpu/0  2.80  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:44|main      Speed   3.15 Mh/s    gpu/0  3.15  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:44|main      Speed   3.15 Mh/s    gpu/0  3.15  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:45|main      Speed   2.94 Mh/s    gpu/0  2.94  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:45|main      Speed   2.94 Mh/s    gpu/0  2.94  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:46|main      Speed   2.80 Mh/s    gpu/0  2.80  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:47|main      Speed   2.80 Mh/s    gpu/0  2.80  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:47|main      Speed   3.00 Mh/s    gpu/0  3.00  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:48|main      Speed   2.89 Mh/s    gpu/0  2.89  [A0+0:R0+0:F0] Time: 00:00
  m  15:57:48|main      Speed   2.89 Mh/s    gpu/0  2.89  [A0+0:R0+0:F0] Time: 00:00
```

**Comment by smurfy** [comment]
Benchmark uses a much lower difficulty by default than real mining.
Use ``-M  4776400``  for example, this way you should get similiar results.

**Comment by bmatthewshea** [comment]
I have a similar issue and have found no answer after extensive searching - driving me nuts. This was the only place I could think to write this up.

I have a Sapphire RX580-8gb w/ Hynix.
For some reason when I mine Ethereum (or ETC), I get a lower hashrate (ethminer output) than compared to other coins based on geth/ethash.
For instance, I get about 22MH if mining Ethereum with ethminer but I get 29MH+ if I mine pirl/or many other coins.

My card is overclocked and bios modded (I got 15mh out of box against ETH/ethminer). :(

Other miners (equi, cnight, etc) seem to be same as other 580 benchmarks I see. Just ethminer versus ETH.

I would chalk this up to mining/chain difficulty as well, except it's such a big difference and my friends do not see this difference at all - on any of their cards - when using ethminer under same. Personally, I also do not have this problem on my NV 1060 6gb - it reports 24MH no matter the 'coin' being mined.
Am I completely missing something?
Thanks

**Comment by DeadManWalkingTO** [comment]
1. Try Update Drivers.
Download and install the latests.

2. Try use -U for CUDA devices.
That's my launch command:
`ethminer.exe -RH -U -S eu1.ethermine.org:4444 -FS us1.ethermine.org:4444 -O 0x7013275311fc37ccc1e40193D75086293eCb43A4.test`

3. Try to change P2 State and Power managment mode.
You can use [NVidiaProfileInspectorDmW](https://github.com/DeadManWalkingTO/NVidiaProfileInspectorDmW).
For the best mining hashrate choose from sector "5 - Common":
    - CUDA - Force P2 State (Set to "Off")
    - Power managment mode (Set to "Prefer maximum performance")

4. Try Tweak Win10.
You can use [Windows10MiningTweaksDmW](https://github.com/DeadManWalkingTO/Windows10MiningTweaksDmW) (#695).

5. Try Optimize/Overclock GPUs. 
You can use [MSI Afterburner](https://www.msi.com/page/afterburner) for GPU OverClock/Optimize. 

6. Try use a WatchDog
You can use [ETHminerWatchDogDmW](https://github.com/DeadManWalkingTO/ETHminerWatchDogDmW) (#735).

Please feedback.
Thank you!

**Comment by muhfugen** [comment]
@DeadManWalkingTO, your suggestions are absolutely retarded and you should feel bad for posting them. When you go to bed tonight, I hope you cry yourself to sleep over how much you've embarrassed yourself. You seem like you're the type of moron who would buy "Registry Cleaner" software and religiously run it every month under the idea it is in some way beneficial.

For example in Windows10MiningTweaksDmW, it does shit like disable My Computer from the desktop, displaying file extensions, and set Internet Explorer's search engine to google. Please kill yourself you fucking faggot.

**Comment by bmatthewshea** [comment]
Just a note on my AMD resolution:
I resolved my 580 issue by finally getting the AMD blockchain driver "working".
Turned out Windows 7 + "Windows 7" BC driver was the problem. As in: **it doesn't work**. (Which is why I had been using the production driver)

I installed Windows 10 and the Win10 version of the BC driver and hashrate is now around 28-29MH on ETH/ETC (meaning I get same speed now on those as I was getting on other alt coins previously).
